ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -913,7 +913,35 @@ public func zip<S: SequenceType, T, Error where S.Generator.Element == SignalPro
 	return .empty
 }
 
+extension SignalProducer where T: SignalProducerType, E == T.E {
+	/// Flattens the inner producers sent upon `producer` (into a single producer of
+	/// values), according to the semantics of the given strategy.
+	///
+	/// If `producer` or an active inner producer emits an error, the returned
+	/// producer will forward that error immediately.
+	///
+	/// `Interrupted` events on inner producers will be treated like `Completed`
+	/// events on inner producers.
+	@warn_unused_result(message="Did you forget to call `start` on the producer?")
+	public func flatten(strategy: FlattenStrategy) -> SignalProducer<T.T, E> {
```

Comment:
ps practice of breaking down into smaller steps with an expected type clearly defined is a good one, I use it often as well. The problem in my case that there was no further way to breakdown... I actually found the place of an error but because of a very complex signature of signals etc was too blind to notice it.